src/scion/math/InterpolationTable/src/ctor.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+ *  @brief Copy constructor
+ *
+ *  @param[in] table    the table to be copied
+ */
+InterpolationTable( const InterpolationTable& table ) :
+  x_( table.x_ ), y_( table.y_ ),
+  boundaries_( table.boundaries_ ),
+  interpolants_( table.interpolants_ ) {
+
+  this->generateTables();
+}
+
+/**
+ *  @brief Move constructor
+ *
+ *  @param[in] table    the table to be moved
+ */
+InterpolationTable( InterpolationTable&& table ) :
+  x_( std::move( table.x_ ) ), y_( std::move( table.y_ ) ),
+  boundaries_( std::move( table.boundaries_ ) ),
+  interpolants_( std::move( table.interpolants_ ) ) {
+
+  this->generateTables();
+}
+
+/**
+ *  @brief Copy assignment using a continuous energy table
+ *
+ *  @param[in] table    the continuous energy table to be copied
+ */
+InterpolationTable& operator=( const InterpolationTable& base ) {
+
+  new (this) InterpolationTable( base );
+  return *this;
+}
+
+/**
+ *  @brief Move assignment using a continuous energy table
+ *
+ *  @param[in] table    the continuous energy table to be moved
+ */
+InterpolationTable& operator=( InterpolationTable&& base ) {
```

Comment:
Copying yourself can be problematic, you may be copying and assigning yourself.

Reply:
Yes, I should fix that at some point. I had noticed this one already but I did not fix it yet.